### PR TITLE
Add the SES Barclamp

### PR DESCRIPTION
**Pull request by hemna**
This padds the new SUSE Enterprise Storage (SES) barclamp.
This barclamp can accept the SES 5.5 openstack runner's
config.yml output.  The config.yml is added to a UI textbox
and then parsed into the required fields in the barclamp ui,
to support integration with an external SES/ceph cluster.

TODO:
  The only portion of the barclamp that currently works is the UI.
  Still have to add the required service node's system level packages.
  Still have to add the libvirt secret on all of the nova compute nodes.
  Still have to make the required changes to support the ceph backend
  settings in each of the service config files (cinder.conf, etc.)

### CODE REVIEW

File: crowbar_framework/app/assets/javascripts/barclamps/ses/application.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2011-2013, Dell
```
Comment (dirkmueller) [inline crowbar_framework/app/assets/javascripts/barclamps/ses/application.js]: Really?

File: crowbar_framework/app/assets/javascripts/barclamps/ses/application.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * Copyright 2013-2014, SUSE LINUX Products GmbH
```
Comment (dirkmueller) [inline crowbar_framework/app/assets/javascripts/barclamps/ses/application.js]: That should be SUSE Linux GmbH

File: crowbar_framework/app/models/ses_service.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  base["attributes"][@bc_name]["nova_instance"] = find_dep_proposal("nova")
+  base["attributes"][@bc_name]["keystone_instance"] = find_dep_proposal("keystone")
+  base["attributes"][@bc_name]["glance_instance"] = find_dep_proposal("glance")
+
+  @logger.debug("SES create_proposal: exiting")
+  base
+end
+
+def validate_proposal_after_save(proposal)
+  validate_one_for_role proposal, "ses-controller"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ses_service.rb]: why do we need a ses-controller role? wouldn't that be better a ceph-client role, autoassigned to all?

Comment (hemna) [reply crowbar_framework/app/models/ses_service.rb]: I don't think we need one, we were just following what little to non-existent, documentation there was on creating a new barclamp and the purposes of each component.  It's completely unclear on the purpose of a lot of what comprises a barclamp.   I'm not even sure what's needed here at all.
The more I look into the rest of the codebase, I'm not sure what the ses barclamp will need to do other than collect the ceph config from the yaml at this point.

**Comment by hemna** [comment]
The current pull request is a WIP that has lots of formatting/syntax issues.   The branch is for sharing code between myself and Charles and for a light review.  This is currently not intented to land in the main repo.

File: crowbar_framework/app/views/barclamp/ses/_edit_attributes.html.haml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    %fieldset
+      %legend
+        = t(".yaml_header")
+
+      = text_field %w(yaml_file)
+
+      %div.form-group.pull-left
+        %input(id='parse_yaml' class="form-control" type="button" value="Parse Yaml")
+        %div#yaml_success{ :style => "display: none" }
```
[the remark this answers is not in the document]
Comment (hemna) [reply crowbar_framework/app/views/barclamp/ses/_edit_attributes.html.haml]: This parse_yaml translates to access from js.  js can't have names that have dashes in them.

File: crowbar_framework/app/views/barclamp/ses/_edit_attributes.html.haml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    %script#yaml{ :type => "text/javascript", :src => "/assets/barclamps/ses/yaml.min.js" }
+
+    %fieldset
+      %legend
+        = t(".yaml_header")
+
+      = text_field %w(yaml_file)
+
+      %div.form-group.pull-left
+        %input(id='parse_yaml' class="form-control" type="button" value="Parse Yaml")
```
[the remark this answers is not in the document]
Comment (hemna) [reply crowbar_framework/app/views/barclamp/ses/_edit_attributes.html.haml]: This parse_yaml translates to access from js.  js can't have names that have dashes in them.

File: crowbar_framework/app/assets/javascripts/barclamps/ses/yaml.min.js [hunk 6]
```diff
@@ -0,0 +1 @@
...null}}else{o=e;r=o("fs");if(n!=null){return r.readFile(t,function(e,t){if(e){return n(null)}else{return n(String(t))}})}else{i=r.readFileSync(t);if(i!=null){return String(i)}return null}}};return t}();t.exports=r},{"./Pattern":8}],11:[function(e,t,n){var i,r,s,l;r=e("./Parser");i=e("./Dumper");s=e("./Utils");l=function(){function e(){}e.parse=function(e,t,n){if(t==null){t=false}if(n==null){n=null}return(new r).parse(e,t,n)};e.parseFile=function(e,t,n,i){var r;if(t==null){t=null}if(n==null){n=false}if(i==null){i=null}if(t!=null){return s.getStringFromFile(e,function(e){return function(r){var s;s=null;if(r!=null){s=e.parse(r,n,i)}t(s)}}(this))}else{r=s.getStringFromFile(e);if(r!=null){return this.parse(r,n,i)}return null}};e.dump=function(e,t,n,r,s){var l;if(t==null){t=2}if(n==null){n=4}if(r==null){r=false}if(s==null){s=null}l=new i;l.indentation=n;return l.dump(e,t,0,r,s)};e.stringify=function(e,t,n,i,r){return this.dump(e,t,n,i,r)};e.load=function(e,t,n,i){return this.parseFile(e,t,n,i)};return e}();if(typeof window!=="undefined"&&window!==null){window.YAML=l}if(typeof window==="undefined"||window===null){this.YAML=l}t.exports=l},{"./Dumper":1,"./Parser":7,"./Utils":10}]},{},[11]);
```
[the remark this answers is not in the document]
Comment (hemna) [reply crowbar_framework/app/assets/javascripts/barclamps/ses/yaml.min.js]: This is minified js

**Pull request closed without merging**